contracts/ERC20SimpleSwap.sol

```diff
@@ -85,7 +79,7 @@ contract ERC20SimpleSwap {
   }
 
   // recover a signature with the EIP712 signing scheme
-  function recoverEIP712(bytes32 hash, bytes memory sig) internal pure returns (address) {
```

Comment:
Could you please explain this change?

Reply:
querying the chainid is no longer considered pure in 0.8. as this calls `domain()`,  this function cannot be pure either.